### PR TITLE
#656: added HVALS command

### LINKED ISSUE #656: Add support for command `HVALS`
Add support for the `HVALS` command in DiceDB similar to the [`HVALS` command in Redis](https://redis.io/docs/latest/commands/command-list/). Please refer to the following commit in Redis to understand the implementation specifics - [source](https://github.com/redis/redis/tree/f60370ce28b946c1146dcea77c9c399d39601aaa).

Write unit and integration tests for the command referring to the tests written in the [Redis codebase 7.2.5](https://github.com/redis/redis/tree/f60370ce28b946c1146dcea77c9c399d39601aaa). For integration tests, you can refer to the `tests` folder. Note: they have used `TCL` for the test suite, and we need to port that to our way of writing integration tests using the relevant helper methods. Please refer to our [tests directory](https://github.com/DiceDB/dice/tree/master/tests).

For the command, benchmark the code and measure the time taken and memory allocs using [`benchmem`](https://blog.logrocket.com/benchmarking-golang-improve-function-performance/) and try to keep them to the bare minimum.

### PR DESCRIPTION
**Pull request by bijuli74**
fixes #656

### CODE REVIEW

**Comment by JyotinderSingh** [comment]
Please resolve conflicts

**Comment by bijuli74** [comment]
> Please resolve conflicts

Quick question. Do I have to do anything. I am not able to merge since I got no write access. Thanks.

**Comment by KaviiSuri** [comment]
> Quick question. Do I have to do anything. I am not able to merge since I got no write access. Thanks.
@bijuli74 , you'll have to pull from the `master` branch on dice db and resolve any conflicts that show up. Merging is not possible without that

File: internal/eval/eval.go [hunk 1]
```diff
@@ -2768,6 +2903,26 @@ func evalHSET(args []string, store *dstore.Store) []byte {
 	return clientio.Encode(numKeys, false)
 }
 
+func evalHSETNX(args []string, store *dstore.Store) []byte {
```
Comment (princejha95) [inline internal/eval/eval.go:2906]: Please add documentation for the function

File: internal/eval/eval.go [hunk 2]
```diff
@@ -2809,6 +2964,7 @@ func evalHGET(args []string, store *dstore.Store) []byte {
 	return val
 }
 
+<<<<<<< HEAD
```
Comment (princejha95) [inline internal/eval/eval.go:2967]: I believe you pushed it by mistake. Please remove

**Comment by JyotinderSingh** [comment]
Please fix this PR. It looks like unrelated commits are showing up as a part of this change. This usually happens when the rebase is messed up.

**Comment by JyotinderSingh** [comment]
I've fixed it up for you. Just ran the following:
```bash
git merge master
git push
```

**Pull request merged**